metasynth/distribution/base.py

```diff
@@ -150,53 +149,42 @@ def default_distribution(cls) -> BaseDistribution:
         return cls()
 
 
-class CoreDistribution():  # pylint: disable=too-few-public-methods
-    """Distributions belonging to the core set."""
-
-    privacy = "none"
-    provenance = "builtin"
-
-
-class CategoricalDistribution(BaseDistribution):
-    """Base Class for categorical distributions."""
-
-    var_type = "categorical"
-
-
-class DiscreteDistribution(BaseDistribution):
-    """Base Class for discrete distributions."""
-
-    var_type = "discrete"
-
-
-class ContinuousDistribution(BaseDistribution):
-    """Base Class for continuous distributions."""
-
-    var_type = "continuous"
-
-
-class StringDistribution(BaseDistribution):
-    """Base Class for string distributions."""
-
-    var_type = "string"
-
-
-class DateTimeDistribution(BaseDistribution):
-    """Base Class for date-time distributions."""
-
-    var_type = "datetime"
-
-
-class DateDistribution(BaseDistribution):
-    """Base Class for date distributions."""
-
-    var_type = "date"
-
-
-class TimeDistribution(BaseDistribution):
-    """Base Class for time distributions."""
-
-    var_type = "time"
+def distclass(implements: Optional[str] = None, provenance: Optional[str] = None,
```

Comment:
I think the name `distclass` is not descriptive enough here and elsewhere, how about  `metadist` or even `metasynthdistribution`?

So 
```py
@metadist(implements="core.multinoulli", var_type="categorical")
class MultinoulliDistribution(BaseDistribution):
  ...
```

Conceptually, in my mind this decorator makes sure that any distribution that is at least a BaseDistribution interfaces well with the Metasynth package by implementing some additional properties. So it's a metasynth-specific thing, hence `metadist`. But if you have other thoughts feel free to ignore!

Reply:
Let's do metadist, I think that fits the package, and is not too long.